Number_types/include/CGAL/boost_mp.h

```diff
@@ -0,0 +1,537 @@
+: AST_boost_mp <boost::multiprecision::number<Backend, Eto> > {};
+template <class T1,class T2,class T3,class T4,class T5>
+struct Algebraic_structure_traits<boost::multiprecision::detail::expression<T1,T2,T3,T4,T5> >
+: Algebraic_structure_traits<typename boost::multiprecision::detail::expression<T1,T2,T3,T4,T5>::result_type > {};
+
+template <class Backend, boost::multiprecision::expression_template_option Eto>
+struct Real_embeddable_traits<boost::multiprecision::number<Backend, Eto> >
+: RET_boost_mp <boost::multiprecision::number<Backend, Eto> > {};
+template <class T1,class T2,class T3,class T4,class T5>
+struct Real_embeddable_traits<boost::multiprecision::detail::expression<T1,T2,T3,T4,T5> >
+: Real_embeddable_traits<typename boost::multiprecision::detail::expression<T1,T2,T3,T4,T5>::result_type > {};
+
+template <class Backend, boost::multiprecision::expression_template_option Eto>
+struct Modular_traits<boost::multiprecision::number<Backend, Eto> >
+: MT_boost_mp <boost::multiprecision::number<Backend, Eto> > {};
+template <class T1,class T2,class T3,class T4,class T5>
+struct Modular_traits<boost::multiprecision::detail::expression<T1,T2,T3,T4,T5> >
+: Modular_traits<typename boost::multiprecision::detail::expression<T1,T2,T3,T4,T5>::result_type > {};
+
+template <class Backend, boost::multiprecision::expression_template_option Eto>
+struct Fraction_traits<boost::multiprecision::number<Backend, Eto> >
+: FT_boost_mp <boost::multiprecision::number<Backend, Eto> > {};
+template <class T1,class T2,class T3,class T4,class T5>
+struct Fraction_traits<boost::multiprecision::detail::expression<T1,T2,T3,T4,T5> >
+: Fraction_traits<typename boost::multiprecision::detail::expression<T1,T2,T3,T4,T5>::result_type > {};
+
+template <class Backend, boost::multiprecision::expression_template_option Eto>
+struct Split_double<boost::multiprecision::number<Backend, Eto> >
+: SD_boost_mp <boost::multiprecision::number<Backend, Eto> > {};
+template <class T1,class T2,class T3,class T4,class T5>
+struct Split_double<boost::multiprecision::detail::expression<T1,T2,T3,T4,T5> >
+: Split_double<typename boost::multiprecision::detail::expression<T1,T2,T3,T4,T5>::result_type > {};
+
+
+// Coercions
+
+namespace internal { namespace boost_mp { BOOST_MPL_HAS_XXX_TRAIT_DEF(type); } }
+
+template <class B1, boost::multiprecision::expression_template_option E1, class B2, boost::multiprecision::expression_template_option E2>
+struct Coercion_traits<boost::multiprecision::number<B1, E1>, boost::multiprecision::number<B2, E2> >
+{
+  typedef boost::common_type<boost::multiprecision::number<B1, E1>, boost::multiprecision::number<B2, E2> > CT;
+  typedef Boolean_tag<internal::boost_mp::has_type<CT>::value> Are_implicit_interoperable;
+  // FIXME: the implicit/explicit answers shouldn't be the same...
+  typedef Are_implicit_interoperable Are_explicit_interoperable;
+  // FIXME: won't compile when they are not interoperable.
```

Comment:
The first one doesn't seem very important. CGAL thinks that some types are not interoperable while possibly they could be with explicit casts. It would probably be possible to check if either is explicitly constructible from the other (not sure which one to pick if both conversions work) but I haven't missed such conversions yet.
The second one is there as a reminder in case a compile error ever points to this line. It wouldn't be hard to fix, but it would make the code less readable, so I'd rather not do it unless it actually causes a problem.

Reply:
I will trust your judgment on that. I just wanted to point those `FIXME` out, in case it was something you had forgotten.